### PR TITLE
Improve UI density and code

### PR DESCRIPTION
**Pull request by jwmza**
- [x] Improves `TextEditor` styling
- [x] Cleans up `NewProcess` layout and styling
- [x] Implements `MonoPanel` with improved styling and optional padding across all relevant pages
- [x] Cleans up repetitive and redundant page-level code for layout and spacing
- [x] Clean up repetitive panel-title heading styling
- [x] Create and use `TwoPanel` component for e.g. workspace page
- [ ] Extract excessive page-level styling and components to `/components`

### CODE REVIEW

File: gui/src/components/Panel.svelte [hunk 1]
```diff
@@ -0,0 +1,80 @@
+<script lang="ts">
+  import IconButton from './IconButton.svelte';
+  import BackSVG from './mono/leftarrow.svelte';
+  import * as router from 'svelte-spa-router';
+
+  export let title: string | undefined = undefined;
```
Comment (brandonbloom) [inline gui/src/components/Panel.svelte:6]: There is a shorthand for this:

```typescript
  export let title: string;
```

However, in this case, I think you want:

```typescript
export let title: string | null = null;
```

The reason being that you generally want to avoid using `undefined` as a value. In this case in particular, the goal is to convey "the title is set to none", not "the title has not yet been set".

File: gui/src/components/Panel.svelte [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  export let backRoute: string | undefined = undefined;
+</script>
+
+<div class="panel">
+  {#if $$slots.actions || title !== undefined || backRoute !== undefined}
```
Comment (brandonbloom) [inline gui/src/components/Panel.svelte:11]: If you take the above advice, this becomes `title !== null`. However, a generally more robust check might be `title != null` which checks for == null || == undefined.

Also, while I generally prefer explicit boolean checks rather than relying on "truthiness", it's actually really common and convenient in UI code. So you could even just shorten this to `|| title || backRoute` and drop the null/undefined checks altogether. In that case, you could also change the export to `export let title: string = '';`

File: gui/src/components/Panel.svelte [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    <header>
+      <div class="header-title">
+        {#if backRoute !== undefined}
```
Comment (brandonbloom) [inline gui/src/components/Panel.svelte:14]: Again, in template code, really convenient to just write `{#if backRoute}` even tho it's less explicit, it's basically _why_ JavaScript has this sort of truthiness checks in the first place.

File: gui/src/pages/NewProcess.svelte [hunk 4]
```diff
@@ -48,11 +48,21 @@
       arguments: args,
     });
   };
+
+  const codeExample = `# Export environment variables.
```
Comment (brandonbloom) [inline gui/src/pages/NewProcess.svelte:52]: heh, why didn't i think of this? :-P

**Review by brandonbloom — APPROVED** [review]
LGTM besides some nitpicks.

**Comment by jwmza** [comment]
Thanks @brandonbloom will do

**Comment by jwmza** [comment]
Also now wraps log messages 

image

**Comment by brandonbloom** [comment]
@jwmza you're starting to see some merge conflicts... getter to get smaller changes to the main branch as you go

**Comment by jwmza** [comment]
Yeah I'm moving to merge this ASAP @brandonbloom

**Comment by jwmza** [comment]
Will follow up with componentization from complex page UIs later.
